log-collector-script/eks-log-collector.sh

```diff
@@ -0,0 +1,566 @@
+      cp --force /var/log/dmesg "${COLLECT_DIR}/kernel/dmesg.boot"
+  fi
+  dmesg > "${COLLECT_DIR}/kernel/dmesg.current"
+  dmesg --ctime > "${COLLECT_DIR}/kernel/dmesg.human.current"
+  uname -a > "${COLLECT_DIR}/kernel/uname.txt"
+
+  ok
+}
+
+get_docker_logs() {
+  try "collect Docker daemon logs"
+
+  case "${INIT_TYPE}" in
+    systemd)
+      journalctl --unit=docker --since "${DAYS_10}" > "${COLLECT_DIR}"/docker/docker.log
+      ;;
+    other)
+      for entry in docker upstart/docker; do
+        if [[ -e "/var/log/${entry}" ]]; then
+          cp --force --recursive --dereference /var/log/"${entry}" "${COLLECT_DIR}"/docker/
+        fi
+      done
+      ;;
+    *)
+      warning "The current operating system is not supported."
+      ;;
+  esac
+
+  ok
+}
+
+get_k8s_info() {
+  try "collect kubelet information"
+
+  if [[ -n "${KUBECONFIG:-}" ]]; then
+    command -v kubectl > /dev/null && kubectl get --kubeconfig=${KUBECONFIG} svc > "${COLLECT_DIR}"/kubelet/svc.log
+    kubectl --kubeconfig=${KUBECONFIG} config view  --output yaml > "${COLLECT_DIR}"/kubelet/kubeconfig.yaml
+
+  elif [[ -f /etc/systemd/system/kubelet.service ]]; then
+    KUBECONFIG=`grep kubeconfig /etc/systemd/system/kubelet.service | awk '{print $2}'`
+    command -v kubectl > /dev/null && kubectl get --kubeconfig=${KUBECONFIG} svc > "${COLLECT_DIR}"/kubelet/svc.log
+    kubectl --kubeconfig=${KUBECONFIG} config view  --output yaml > "${COLLECT_DIR}"/kubelet/kubeconfig.yaml
+
+      if [[ $? != 0 ]]; then
+         if [[ -f /etc/eksctl/kubeconfig.yaml ]]; then
+            KUBECONFIG="/etc/eksctl/kubeconfig.yaml"
+            command -v kubectl > /dev/null && kubectl get --kubeconfig=${KUBECONFIG} svc > "${COLLECT_DIR}"/kubelet/svc.log
+            kubectl --kubeconfig=${KUBECONFIG} config view  --output yaml > "${COLLECT_DIR}"/kubelet/kubeconfig.yaml
+         fi
+      fi
+
+  elif [[ -f /etc/eksctl/kubeconfig.yaml ]]; then
+    KUBECONFIG="/etc/eksctl/kubeconfig.yaml"
+    command -v kubectl > /dev/null && kubectl get --kubeconfig=${KUBECONFIG} svc > "${COLLECT_DIR}"/kubelet/svc.log
+    kubectl --kubeconfig=${KUBECONFIG} config view  --output yaml > "${COLLECT_DIR}"/kubelet/kubeconfig.yaml
+
+  else
+    echo "======== Unable to find KUBECONFIG, IGNORING POD DATA =========" >> "${COLLECT_DIR}"/kubelet/svc.log
+  fi
+
+  case "${INIT_TYPE}" in
+    systemd)
+      timeout 75 journalctl --unit=kubelet --since "${DAYS_10}" > "${COLLECT_DIR}"/kubelet/kubelet.log
+      timeout 75 journalctl --unit=kubeproxy --since "${DAYS_10}" > "${COLLECT_DIR}"/kubelet/kubeproxy.log
+
+      for entry in kubelet kube-proxy; do
+        systemctl cat "${entry}" > "${COLLECT_DIR}"/kubelet/"${entry}"_service.txt 2>&1
+      done
+      ;;
+    *)
+      warning "The current operating system is not supported."
+      ;;
+  esac
+
+  ok
+}
+
+get_ipamd_info() {
+  try "collect L-IPAMD information"
+  if [[ "${ignore_introspection}" == "false" ]]; then
```

Comment:
Hmm, shouldn't this be checking that it is not true? Currently, it's off by default it seems.

Reply:
I have set ignore_introspection, ignore_metrics parameters default to [false](https://github.com/awslabs/amazon-eks-ami/blob/f6155ab958b630d7f8a45a13787d0dd2073481a5/log-collector-script/eks-log-collector.sh#L34-L37).